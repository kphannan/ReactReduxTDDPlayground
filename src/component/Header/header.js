import React, { Component } from 'react';

export default class Header extends Component {

  render() {
    return (
      <div className="header">
            <div className="header-logo">
              <i className="icon_homedepot"></i>
            </div>

             <div className="header-info">
               <div className="product-info">W/MRI</div>
               <div className="user-info">User Name | Store #0000 | Atlanta, GA</div>
           </div>
            <div className="header-search">
                <input type="search" placeholder="Search Here"/>
                <i className="icon_search"></i>
            </div>
            <div className="header-actions">
                <div className="active"><i class="icon_bell"></i>Active Link</div>
                <div><i className="icon_cart"></i>Nav Link</div>
                <div><i className="icon_exit-to-app"></i>Nav Link</div>
            </div>
         </div>
    );
  }
}


             // <div className="header-info">
             //     <label className="product-info">W/MRI</label>
             // </div>

