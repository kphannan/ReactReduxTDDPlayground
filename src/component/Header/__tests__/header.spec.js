import React from 'react'
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import Header from '../header'

Enzyme.configure({ adapter: new Adapter() })

describe('Header Testing', () => {

  const component = shallow(<Header />)

  test('verify header title', () => {
    expect(component.contains('W/MRI')).toBe(true)
  })

  test('verify against header snapshot', () => {
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  test('verify css header matches UX style guide header class', () => {
    expect(component.containsMatchingElement('.header'));
  })
})
