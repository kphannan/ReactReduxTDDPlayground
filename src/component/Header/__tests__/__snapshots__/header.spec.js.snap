// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Testing verify against header snapshot 1`] = `
<div
  className="header"
>
  <div
    className="header-logo"
  >
    <i
      className="icon_homedepot"
    />
  </div>
  <div
    className="header-info produc-info"
  >
    W/MRI
  </div>
</div>
`;
